File: C/GuessingGame/guessingGame.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int testGuess(int guess, int actual)
{
	if ( guess < actual) return -1;

	else if (guess > actual) return 1;

	else return 0;
}

int main(int argc, char** argv)
{
	int max;
	printf("Choose a max from 1: ");
	scanf("%d", &max);

	srand(time(NULL));
	int actual = rand() % (max);
	
	int maxTries  = 10;
	int tries = 0;
	while (tries <= maxTries)
	{
		printf("Pick a number between 1 and %d: ", max);
		int guess;
		scanf("%d", &guess);
		
		int status = testGuess(guess, actual);
		switch (status)
		{
			case -1:
				printf("Too Low!\n");
				tries++;
				break;
			case 1:
				printf("Too High!\n");
				tries++;
				break;
			case 0:
				printf("You guessed correct!\n");
				return 0;
		}
	}
	printf("You Lost! Exceeded number of tries!");
	return 0;
}

File: Python/GuessingGame/guessingGame.py
import random


def testGuess(guess: int, actual: int) -> int:

    if (actual > guess):
        return -1
    elif (actual < guess):
        return 1
    else:
        return 0


def main() -> int:
    max = int(input("Choose max from 1: "))
    actual = random.randint(1, max)

    maxTries = 10
    tries = 0
    while (tries <= maxTries):
        guess = int(input("Pick a number "))
        status = testGuess(guess, actual)
        match status:
            case 1:
                print("You guessed High!")
                tries += 1
            case -1:
                print("You guessed Low!")
                tries += 1
            case 0:
                print("You win!")
                return 0
    print("You Lost! Max number of tries exceeded!")
    return 1

if __name__ == "__main__":
    main()
